Application/src/main/java/practice/application/service/OrderService.java

```diff
@@ -22,45 +29,67 @@ public class OrderService {
+    public List<OrderResponseDTO> getOrders(MemberEntity member, OrderStatus orderStatus){
+        List<OrderEntity> orderEntityList = orderRepository.findByMemberAndStatus(member, orderStatus)
+                .orElseThrow(() -> new NotFoundException("해당 유저에 대한 주문은 없습니다"));
 
         if(orderEntityList.isEmpty()){
-          throw new NotFoundException("해당 이메일에 대한 주문은 없습니다");
+            throw new NotFoundException("해당 유저에 대한 주문은 없습니다");
         }
 
 
-        return orderEntityList;
+        return orderEntityList.stream().map(orderEntity -> new OrderResponseDTO(orderEntity)).collect(Collectors.toList());
     }
 
     @Transactional
-    public OrderEntity cancelOrder(String orderId) {
-        OrderEntity orderEntity = orderRepository.findFetchById(orderId).orElseThrow(() -> new NotFoundException("해당 주문을 찾을 수 없습니다"));
-
-        orderEntity.orderCancel();
-
+    public OrderEntity save(MemberEntity member, OrderCreateDTO orderCreateDTO) {
+        List<OrdersItemEntity> orderItems = orderItemsService.createOrderItems(orderCreateDTO.getOrderItemsDTOS());
 
-        return orderEntity;
+        OrderEntity orderEntity = checkExistOrder(member, orderCreateDTO, orderItems);
 
+        return orderRepository.save(orderEntity);
     }
 
+    @Transactional
+    public PatchOrderStatusDTO paymentOrder(String orderId) {
+        OrderEntity orderEntity = orderRepository.findById(orderId).orElseThrow(() -> new NotFoundException("해당 주문을 찾을 수 없습니다"));
 
+        orderEntity.changeStatusPayment();
 
+        return new PatchOrderStatusDTO(orderEntity.getStatus());
+    }
 
+    @Transactional
+    public PatchOrderStatusDTO cancelOrder(String orderId) {
+        OrderEntity orderEntity = orderRepository.findFetchById(orderId).orElseThrow(() -> new NotFoundException("해당 주문을 찾을 수 없습니다"));
 
+        if(orderEntity.getStatus().equals(OrderStatus.DELIVERED)) {
+            throw new ImpossibleCancelException();
+        }
 
+        orderEntity.orderCancel();
 
+        return new PatchOrderStatusDTO(orderEntity.getStatus());
+    }
 
+    /**
+     * Member와 status가 RESERVED인 Order를 조회합니다.
+     * 있을 경우 해당 Order를 사용하여 추가로 상품을 주문할 수 있습니다.
+     * 없을 경우 새로운 Order를 생성합니다.
+     */
+    public OrderEntity checkExistOrder(MemberEntity member,
+                                       OrderCreateDTO orderCreateDTO,
+                                       List<OrdersItemEntity> orderItems) {
+        Optional<OrderEntity> optionalOrder = orderRepository.findByMemberAndReservedStatus(member);
+
+        if(optionalOrder.isPresent()) {
+            OrderEntity orderEntity = optionalOrder.get(); // 찾은 Order 사용
+            orderEntity.addOrderItems(orderItems);
+
+            return orderEntity;
+        } else {
+            // 새로운 Order 생성
+            return new OrderEntity(member, orderCreateDTO.getEmail(), orderCreateDTO.getPostCode(), orderItems);
+        }
+    }
```

Comment:
`orderItems` 대신 `newOrderItems`?

```suggestion
    public OrderEntity checkExistOrder(MemberEntity member,
                                       OrderCreateDTO orderCreateDTO,
                                       List<OrdersItemEntity> newOrderItems) {
        Optional<OrderEntity> optionalOrder = orderRepository.findByMemberAndReservedStatus(member);

        if(optionalOrder.isPresent()) {
            OrderEntity orderEntity = optionalOrder.get(); // 찾은 Order 사용
            orderEntity.addOrderItems(newOrderItems);

            return orderEntity;
        } else {
            // 새로운 Order 생성
            return new OrderEntity(member, orderCreateDTO.getEmail(), orderCreateDTO.getPostCode(), newOrderItems);
        }
    }
```

Reply:
newOrderItems라는 변수명이 더 이해하기 편하겠네요~👍